### PR TITLE
[Model] Add support for xverse

### PR DESCRIPTION
**Pull request by hxer7963**
[Model] Add support for xverse

llama.cpp: Add support for Xverse model architecture
* init_mappings: Turn off prefetching when load xveres model to avoid struck when call mmap on linux os.

gguf-py : Add convert for Xverse model architecture

### CODE REVIEW

File: convert-hf-to-gguf.py [hunk 1]
```diff
@@ -212,6 +212,7 @@ def from_model_architecture(cls, arch):
         try:
             return cls._model_classes[arch]
         except KeyError:
+            print(f"{cls._model_classes}")
```
Comment (cebtenzzre) [inline convert-hf-to-gguf.py:215]: I don't think we need to leave this in. Unless you think the exception message should mention the full list of supported models.

**Comment by hxer7963** [comment]
> > * init_mappings: Turn off prefetching when load xveres model to avoid struck when call mmap on linux os.
> 
> Can you elaborate, why is this necessary specifically for this model and not others?

Nice question!

I just turned on the prefetch switch and retested the xverse model loading process locally, and found that there was no problem.
During previous testing, it was possible that the model was too large (65B model) and the loading process was slow, mistakenly thinking that the program was stuck.

The prefetch settings have been restored in the new submission.

File: llama.cpp [hunk 2]
```diff
@@ -6379,6 +6432,107 @@ struct llm_build_context {
...
+                );
+                cb(Kcur, "Kcur", il);
+
+
+                cur = llm_build_kv(ctx0, model, hparams, kv_self, gf,
+                        model.layers[il].wo, NULL,
+                        Kcur, Vcur, Qcur, KQ_mask, KQ_pos, n_ctx, n_tokens, kv_head, n_kv, 1.0f/sqrtf(float(n_embd_head)), cb, il);
+                cb(cur, "kqv_out", il);
+            }
+
```
Comment (compilade) [inline llama.cpp:6495]: Something to consider when merging the recent changes from `master`:

As of #6122, the unused outputs of the last layers are excluded.

Right before `ffn_inp`, there should be something like this:

```suggestion

            if (il == n_layer - 1) {
                // skip computing output for unused tokens
                struct ggml_tensor * inp_out_ids = build_inp_out_ids();
                cur   = ggml_get_rows(ctx0,   cur, inp_out_ids);
                inpSA = ggml_get_rows(ctx0, inpSA, inp_out_ids);
            }

```

but this is only relevant once the most recent `master` is merged in your branch first to include the changes from #6122.

Comment (hxer7963) [reply llama.cpp:6527]: I have already include the changes from #6122  to exclude the unused outputs of the last layers in latest commit.

**Review by ggerganov — APPROVED** [review]

**Comment by hxer7963** [comment]
Hello, @cebtenzzre @compilade 

I've addressed all the feedback in the pull request.

The EditorConfig Checker and flake8 Lint failed, how can i fix it?

Could you please take another look at the code changes and provide your review?

Thank you for your time and assistance.

Best regards,

hxer7963.

File: llama.cpp [hunk 3]
```diff
@@ -6411,6 +6464,114 @@ struct llm_build_context {
...
+                    n_rot, rope_type, 0, n_orig_ctx, freq_base, freq_scale,
+                    ext_factor, attn_factor, beta_fast, beta_slow
+                );
+                cb(Kcur, "Kcur", il);
+
+
+                cur = llm_build_kv(ctx0, model, hparams, kv_self, gf,
+                        model.layers[il].wo, NULL,
+                        Kcur, Vcur, Qcur, KQ_mask, KQ_pos, n_ctx, n_tokens, kv_head, n_kv, 1.0f/sqrtf(float(n_embd_head)), cb, il);
+                cb(cur, "kqv_out", il);
```
Comment (slaren) [inline llama.cpp:6525]: `kqv_out` is already set in `llm_build_kv`.
```suggestion
```

Comment (hxer7963) [reply llama.cpp]: Thanks for @slaren 's review and suggestions. I have fixed the relevant issues in the new commit.

But the new commit still has 2 checks failed:
- Benchmark/bench server baseline error: Resource not accessible by integration. Does this look like a **permission issue**?
- MacOS latest make x64: error failed to allocate tensors [Metal] ELU (type=f32, ne=[128,10,10,10]): not supported [Metal]...

I'm not sure how to fix these two issues. Can you give me some advice?

Best regards,

hxer7963.

**Comment by ggerganov** [comment]
The macOS runner is occasionally started without access to GPU, so this is not related to this PR. The Benchmark CI was recently added so we can expect some instability at the start - nothing to worry about

**Comment by hxer7963** [comment]
> The macOS runner is occasionally started without access to GPU, so this is not related to this PR. The Benchmark CI was recently added so we can expect some instability at the start - nothing to worry about

Thank you very much for ggerganov's patient reply

May I ask how long it usually takes to merge to the main branch?

**Comment by ggerganov** [comment]
We can merge after @slaren's approval

File: llama.cpp [hunk 4]
```diff
@@ -6411,6 +6464,113 @@ struct llm_build_context {
...
+                cb(Qcur, "Qcur", il);
+
+                Kcur = ggml_rope_custom(
+                    ctx0, ggml_reshape_3d(ctx0, Kcur, n_embd_head, n_head_kv, n_tokens), inp_pos,
+                    n_rot, rope_type, 0, n_orig_ctx, freq_base, freq_scale,
+                    ext_factor, attn_factor, beta_fast, beta_slow
+                );
+                cb(Kcur, "Kcur", il);
+
+
```
Comment (slaren) [inline llama.cpp:6521]:
```suggestion

```

**Review by slaren — APPROVED** [review]

**Pull request merged**